examples/transformerconfigs/README.md

```diff
@@ -119,44 +119,472 @@ commonAnnotations:
 
 ## Name reference transformer
 
-Name reference transformer's configuration is different from all other transformers. It contains a list of `nameReferences`, which represent all of the possible fields that a type could be used as a reference in other types of resources. A `nameReference` contains a type such as ConfigMap as well as a list of `fieldSpecs` where ConfigMap is referenced in other resources. Here is an example:
+`nameReference` Transformer is used to tie a target resource's name to a list of other resources' referrers' names.
+Once tied, the referrers' names will change alongside the target name via transformers like `namePrefix` and `nameSuffix`
```

Comment:
+1 to adding this, I am surprised something so basic and essential wasn't stated in the original doc.